src/Service/ZaakService.php

```diff
@@ -391,148 +165,121 @@ private function checkZaakType(array $case)
         if ($zaakTypeObject === null) {
             return null;
-        }//end if
+        }
 
         return $zaakTypeObject;
 
     }//end checkZaakType()
 
 
     /**
-     * Checks and fetches or creates a Synchronization for this case.
-     *
-     * @param array $case xxllnc case object.
+     * Synchronises a case to zgw zaak based on the data retrieved from the Xxllnc api.
      *
-     * @return Synchronization
-     */
-    private function getSyncForCase(array $case): Synchronization
-    {
-        // Find or create synchronization object.
-        $synchronization = $this->synchronizationService->findSyncBySource($this->xxllncAPI, $this->zaakSchema, $case['reference']);
-        $synchronization->setMapping($this->caseMapping);
-        $synchronization = $this->synchronizationService->synchronize($synchronization, $case);
-
-        return $synchronization;
-
-    }//end getSyncForCase()
-
-
-    /**
-     * Creates ZGW Zaak subobjects.
+     * @param array $case  The case to synchronize.
+     * @param bool  $flush Wether or not the case should be flushed already (not functional at this time)
      *
-     * @param array        $zaakArray      ZGW Zaak.
-     * @param array        $zaakTypeArray  ZGW ZaakType.
-     * @param ObjectEntity $zaakTypeObject ZGW ZaakType object.
-     * @param array        $case           xxllnc case object.
+     * @throws \Twig\Error\LoaderError
+     * @throws \Twig\Error\SyntaxError
      *
-     * @return array
+     * @return ObjectEntity The resulting zaak object.
      */
-    private function createSubObjects(array $zaakArray, array $zaakTypeArray, ObjectEntity $zaakTypeObject, array $case): array
+    public function syncCase(array $case, bool $flush = true): ObjectEntity
     {
-        if (isset($zaakTypeArray['statustypen']) === true && isset($case['instance']['milestone']) === true) {
-            $zaakArray = $this->mapStatus($zaakArray, $zaakTypeArray, $case['instance']['milestone']);
-        }
-
-        if (isset($zaakTypeArray['roltypen']) === true && isset($case['instance']['route']['instance']['role']) === true) {
-            $zaakArray = $this->mapRollen($zaakArray, $zaakTypeArray, $case['instance']['route']['instance']['role']);
-        }
-
-        if (isset($zaakTypeArray['eigenschappen']) === true && isset($case['instance']['attributes']) === true) {
-            $zaakArray = $this->mapEigenschappen($zaakArray, $zaakTypeArray, $zaakTypeObject, $case['instance']['attributes']);
-        }
-
-        return $zaakArray;
+        $zaakSchema  = $this->resourceService->getSchema(
+            'https://vng.opencatalogi.nl/schemas/zrc.zaak.schema.json',
+            'common-gateway/xxllnc-zgw-bundle'
+        );
+        $xxllncAPI   = $this->resourceService->getSource(
+            'https://development.zaaksysteem.nl/source/xxllnc.zaaksysteem.source.json',
+            'common-gateway/xxllnc-zgw-bundle'
+        );
+        $caseMapping = $this->resourceService->getMapping(
```

Comment:
Note:
These resourceService functions can still return NULL, if they do, do you still want to continue this function or return and throw and exception or error/log